src/utils/format-chat-history.ts

```diff
@@ -0,0 +1,111 @@
+import { ChatCompletionMessageParam } from "openai/resources";
+import { Context } from "../types";
+import { StreamlinedComment, StreamlinedComments } from "../types/gpt";
+import { fetchPullRequestDiff } from "./issue";
+
+export function formatChatHistory(
+  context: Context,
+  streamlined: Record<string, StreamlinedComment[]>,
+  specAndBodies: Record<string, string>,
+  linkedPulls: Record<string, boolean>
+) {
+  const convoKeys = Object.keys(streamlined);
+  const specAndBodyKeys = Object.keys(specAndBodies);
+
+  const curIssue = {
+    convo: streamlined[convoKeys[0]],
+    specOrBody: specAndBodies[specAndBodyKeys[0]],
+  };
+
+  let issueSpecBlock = "";
+  issueSpecBlock += createHeader("Project Specification", specAndBodyKeys[0]);
+  issueSpecBlock += createSpecOrBody(curIssue.specOrBody);
+  issueSpecBlock += createFooter("Project Specification");
```

Comment:
Use a `string[]` as the buffer and `.join("\n")` at the end. += is too error prone.